### ISSUE TITLE
ConcurrentModificationException when requesting the database size

### ISSUE THREAD

**Issue by jbilcke** [orientechnologies/orientdb#7149]
### OS: macOS Sierra 10.12.2

My app uses OrientJS  to communicate with OrientDB using the REST API.
Their both run on the same machine, and there is only one single OrientDB instance.

## Expected behavior  
Return the size of the database

## Actual behavior  

Sometimes it return the db size, but sometimes there is:

```
 Error executing request
java.util.ConcurrentModificationException
  at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
  at java.util.ArrayList$Itr.next(ArrayList.java:851)
  at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.getSize(OAbstractPaginatedStorage.java:2347)
  at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sizeDatabase(ONetworkProtocolBinary.java:1069)
  at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:373)
  at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:199)
  at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

## Steps to reproduce  

What my app does is upserting +20k items in the database (using the rest api) while regularly asking for the db size at the same time, in parallel. Sometimes it works, sometimes there is this exception. 

Is that allowed/possible? or is there another way?

**Comment by lvca** [comment]
Fixed in 2.2.x branch. it will be part of the next 2.2.17, but you can already use a build from the branch 2.2.x. Thanks.

**Comment by jbilcke** [comment]
woa! thank you! :)
